packages/demo/src/components/examples/TextInputExamples.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import {InputValidator, TextInput, TextInputProvider, Section} from 'react-vapor';
+
+const nonEmptyValidation: InputValidator = (value: string) => {
+    const isEmpty = !value;
+    const isWhiteSpace = !isEmpty && !value.trim();
+    if (isWhiteSpace) {
+        return {status: 'warning', message: 'A white space is not empty, but is not ideal.'};
+    } else if (isEmpty) {
+        return {status: 'invalid', message: 'Cannot be empty.'};
+    } else {
+        return {status: 'valid'};
+    }
+};
+
+export const TextInputExamples: React.FunctionComponent = () => (
+    <TextInputProvider>
+        <Section>
+            <TextInput
+                required
+                showValidationOnBlur
+                validate={nonEmptyValidation}
+                type="text"
```

Comment:
If its called `TextInput` I would expect `type="text"` to be default and not need to be added as a prop. Or is this just to make it obvious?

Reply:
The goal was to make it known that there are other types. Otherwise people will just use `text` for emails and urls. It works but its better to specify the right type for accessibility reasons.

Reply:
makes sense, thanks